uamqp/client.py

```diff
@@ -639,14 +638,30 @@ def receive_message_batch(self, max_batch_size=None, on_message_received=None, t
         else:
             return batch
 
-        while receiving and not expired and len(batch) < max_batch_size:
-            receiving = self.do_work(batch=max_batch_size)
-            while len(batch) < max_batch_size:
-                try:
-                    batch.append(self._received_messages.get_nowait())
-                    self._received_messages.task_done()
-                except queue.Empty:
-                    break
+        to_receive_size = max_batch_size - len(batch)
+        before_queue_size = self._received_messages.qsize()
```

Comment:
Shouldn't `qsize` always be 0 here?
Otherwise the batch would have been returned on line 639?

Reply:
the `break` at line 637 will break the while loop + else statement, so line 639 won't be executed when encountering with `queue.Empty` error